designer/app.py

```diff
@@ -1115,17 +1142,8 @@ def on_request_close(self, *args):
         if not self._curr_proj_changed:
             self._perform_quit()
             return False
-        self._confirm_dlg = ConfirmationDialog('All unsaved changes will be'
-                                               ' lost.\n'
-                                               'Do you want to quit?')
-        self._confirm_dlg.bind(on_ok=self._perform_quit,
-                               on_cancel=self._cancel_popup)
 
-        self._popup = Popup(title='Quit', content=self._confirm_dlg,
-                            size_hint=(None, None), size=('200pt', '150pt'),
-                            auto_dismiss=False)
-        self._popup.open()
-        return True
```

Comment:
just keep this return True, otherwise the window will be closed.
Your pr is really great, just fix this line so we can merge